Sources/TuistGenerator/Generator/KnownAssetTagsGenerator.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import PathKit
+import TuistGraph
+
+private struct ContentJson: Decodable {
+    struct ContentProperties: Decodable {
+        enum CodingKeys: String, CodingKey {
+            case onDemandResourceTags = "on-demand-resource-tags"
+        }
+
+        let onDemandResourceTags: [String]
+    }
+
+    let properties: ContentProperties
+}
+
+protocol KnownAssetTagsGenerating: AnyObject {
+    func generate(project: Project) throws -> [String]
+}
+
+final class KnownAssetTagsGenerator: KnownAssetTagsGenerating {
+    func generate(project: Project) throws -> [String] {
```

Comment:
This one is a good candidate to be tested. In particular, the following scenario:

- It returns the tags from: initialInstallTags, prefetchOrderTags, asset catalogs,

Reply:
@pepicrft thank you for the approve. Does it means the PR is ready for merge or do you want me to cover this fetched by unit tests?